File: ios/Sources/TaigiKeyboard/Settings/InputMode.swift
// 中文: 輸入模式 (InputMode) 領域型別。POJ / TL / English / TPS 四種。
// 中文: Foundation-only 純值型別,shared-core 候選;displayName 在 SettingsModels.swift 以 extension 提供。

import Foundation

// MARK: - Shared-Core Candidate

// Pure logic, Foundation-only. Eligible for cross-platform extraction.

/// Input mode selected by the user: POJ (Pe̍h-ōe-jī), TL (Tâi-lô),
/// English passthrough, or TPS (Taiwanese Phonetic Symbols).
///
/// The enum is a pure value type over `String` so it can live in shared core;
/// the `displayName` localization sibling stays platform-side and is provided
/// via an extension in `SettingsModels.swift`.
// 中文: 使用者目前選用的輸入法種類。rawValue 與 UserDefaults 持久化欄位對齊。
public enum InputMode: String, CaseIterable {
    // 中文: Pe̍h-ōe-jī 白話字。
    case poj // Pe̍h-ōe-jī
    // 中文: Tâi-lô 教育部臺羅。
    case tl // Tâi-lô
    // 中文: 英文直通模式,不走音標處理。
    case english
    // 中文: TPS 台語注音符號 (Taiwanese Phonetic Symbols)。
    case tps // Taiwanese Phonetic Symbols
}

File: android/app/src/main/java/com/siansiansu/taigikeyboard/ime/core/settings/InputMode.kt
// 拼音輸入模式列舉 — POJ / TL / English 三種(對應 iOS 還多一個 TPS)。
// fromPrefString 把 DataStore 字串("poj"/"tl"/"tps"/"english")轉成 enum;
// "tps" 對應 TL(共用 TL 表)、未知值退回 POJ(對齊舊 KeyView / TextInputManager 預設)。

package com.siansiansu.taigikeyboard.ime.core.settings

/**
 * User-selected romanization input mode.
 *
 * Mirrors iOS `Settings/InputMode.swift`. Pure Kotlin enum — no platform
 * dependencies — suitable for shared-core extraction.
 *
 * Restored 2026-04-27 after `D9.4 commit 9` over-deleted alongside
 * `ToneConverterModels.kt` (which housed phonetic mappings now owned by
 * Rust). The enum itself is a Composing/UI domain concern, not phonetics.
 *
 * Note: iOS adds a fourth case (`tps`) for Taiwanese Phonetic Symbols.
 * Android keeps three cases here to match the original `when` exhaustion
 * across call sites; aligning with iOS is a separate scope.
 */
enum class InputMode {
    /** Pe̍h-ōe-jī (白話字) */
    POJ,

    /** Tâi-lô (台羅) */
    TL,

    /** English passthrough */
    ENGLISH,
    ;

    companion object {
        /** Coerce a stored preference string (`"poj"` / `"tl"` / `"tps"` /
         *  `"english"`) into an [InputMode]. `"tps"` maps to [TL] because
         *  TPS shares the TL phonetic-table path; unknown values fall back
         *  to [POJ] (matches the legacy `KeyView.getComputedLetter` and
         *  `TextInputManager.handleTaigiInput` defaults). */
        fun fromPrefString(value: String): InputMode =
            when (value) {
                "poj" -> POJ
                "tl", "tps" -> TL
                "english" -> ENGLISH
                else -> POJ
            }
    }
}
